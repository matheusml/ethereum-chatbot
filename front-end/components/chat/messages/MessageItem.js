import React from "react";
import { StyleSheet, Text, View } from "react-native";
import PropTypes from "prop-types";

export default class MessageItem extends React.Component {
  render() {
    const { item } = this.props;
    return (
      <View style={styles.row}>
        <Text style={styles.sender}>{item.sender}</Text>
        <Text style={styles.message}>{item.message}</Text>
      </View>
    );
  }
}

MessageItem.propTypes = {
  item: PropTypes.shape({
    sender: PropTypes.string,
    message: PropTypes.string
  })
};

const styles = StyleSheet.create({
  row: {
    flexDirection: "row",
    padding: 20,
    borderBottomWidth: 1,
    borderBottomColor: "#eee"
  },
  message: {
    fontSize: 18
  },
  sender: {
    fontWeight: "bold",
    paddingRight: 10
  }
});
