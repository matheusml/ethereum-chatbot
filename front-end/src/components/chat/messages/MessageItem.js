import React from "react";
import { StyleSheet, Text, View } from "react-native";
import PropTypes from "prop-types";

export default class MessageItem extends React.Component {
  render() {
    const { item } = this.props;
    return (
      <View style={styles.row}>
        <Text style={styles.sender}>{item.sender}</Text>
        <Text style={styles.message}>{item.message}</Text>
      </View>
    );
  }
}

MessageItem.propTypes = {
  item: PropTypes.shape({
    sender: PropTypes.string,
    message: PropTypes.string
  })
};

const styles = StyleSheet.create({
  row: {
    flexDirection: "row",
    padding: 20,
    borderBottomWidth: 1,
    borderBottomColor: "#eee"
  },
  message: {
    fontSize: 16,
    flex: 0.9,
    flexDirection: "column"
  },
  sender: {
    fontWeight: "bold",
    paddingRight: 10
  }
});
