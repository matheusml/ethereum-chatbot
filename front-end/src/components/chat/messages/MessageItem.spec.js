import React from "react";

import MessageItem from "./MessageItem";

import renderer from "react-test-renderer";

describe("MessageItem", () => {
  it("should match regular message snapshot", () => {
    const props = {
      item: {
        sender: "Sender",
        message: "message"
      }
    };
    const tree = renderer.create(<MessageItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match balance message snapshot", () => {
    const props = {
      item: {
        sender: "Sender",
        message: {
          balance: 1
        }
      }
    };
    const tree = renderer.create(<MessageItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match transaction message snapshot", () => {
    const props = {
      item: {
        sender: "Sender",
        message: {
          transaction: {
            blockHash: 0,
            from: 1,
            to: 2,
            value: 3,
            gas: 4
          }
        }
      }
    };
    const tree = renderer.create(<MessageItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
