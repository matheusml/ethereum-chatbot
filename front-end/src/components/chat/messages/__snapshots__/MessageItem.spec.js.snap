// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem should match balance message snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 0.9,
      "flexDirection": "column",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
        "paddingRight": 10,
      }
    }
  >
    Sender
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "fontStyle": "italic",
      }
    }
  >
    Balance: 
    1
  </Text>
</View>
`;

exports[`MessageItem should match regular message snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 0.9,
      "flexDirection": "column",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
        "paddingRight": 10,
      }
    }
  >
    Sender
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    message
  </Text>
</View>
`;

exports[`MessageItem should match transaction message snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 0.9,
      "flexDirection": "column",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
        "paddingRight": 10,
      }
    }
  >
    Sender
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "fontStyle": "italic",
        }
      }
    >
      Block hash: 
      0
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "fontStyle": "italic",
        }
      }
    >
      From: 
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "fontStyle": "italic",
        }
      }
    >
      To: 
      2
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "fontStyle": "italic",
        }
      }
    >
      Value: 
      3
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "fontStyle": "italic",
        }
      }
    >
      Gas: 
      4
    </Text>
  </View>
</View>
`;
